Modules/mathmodule.c

```diff
@@ -2163,6 +2163,26 @@ math_ldexp_impl(PyObject *module, double x, PyObject *i)
     } else {
         errno = 0;
         r = ldexp(x, (int)exp);
+#if _MSC_VER
+        if (r && r > -DBL_MIN && r < DBL_MIN) {
+            /* Denormal result can be incorrectly rounded here (rather,
+               truncated).  Fixed in newer versions of the C runtime, included
+               with Windows 11. */
+            int original_exp;
+            frexp(x, &original_exp);
+            if (original_exp > DBL_MIN_EXP) {
+                /* Shift down to the smallest normal binade.  No bits lost. */
+                x = ldexp(x, DBL_MIN_EXP - original_exp);
+                exp += original_exp - DBL_MIN_EXP;
```

Comment:
Better to add `int shift = DBL_MIN_EXP - original_exp`, then use `shift` in the `ldexp()` call, and change the last line to `exp -= shift;`. I don't care about possibly saving a cycle, but the increased clarity of making it dead obvious that we're adjusting `exp` by the exact amount `ldexp()` moved the bits.